tfe/resource_tfe_oauth_client.go

```diff
@@ -72,13 +80,19 @@ func resourceTFEOAuthClientCreate(d *schema.ResourceData, meta interface{}) erro
 	// Get the organization and provider.
 	organization := d.Get("organization").(string)
 	serviceProvider := tfe.ServiceProviderType(d.Get("service_provider").(string))
+	privateKey := d.Get("private_key").(string)
+
+	if serviceProvider == tfe.ServiceProviderAzureDevOpsServer && privateKey == "" {
+		return fmt.Errorf("PrivateKey is required for ServiceProvider %s", serviceProvider)
```

Comment:
Nit, feel free to ignore: 
I think it might be best to match the casing of the attribute in a TF config, so something like this:
```suggestion
		return fmt.Errorf("private_key is required for service_provider %s", serviceProvider)
```

Reply:
Question: if a private key is specified and service_provider != ado_server, will the server return an error? Or will it just ignore the private key?

Reply:
It will return an error.